### PR TITLE
Add localrc variables to able to change java memory use

### PR DESCRIPTION
**Pull request by safchain**
Add localrc variables in order to change java memory param
of cassandra and zookeeper. Useful for small dev environement.

CASS_MAX_HEAP_SIZE=1G
CASS_HEAP_NEWSIZE=400M

ZK_MAX_HEAP_SIZE=1G
ZK_HEAP_NEWSIZE=400M

### CODE REVIEW

File: lib/neutron_thirdparty/contrail [hunk 1]
```diff
@@ -138,6 +147,7 @@ function install_contrail() {
         cd $CONTRAIL_SRC/third_party
         wget http://apache.mirrors.hoobly.com/zookeeper/stable/zookeeper-3.4.5.tar.gz
         tar xvzf zookeeper-3.4.5.tar.gz
+	test_install_zookeeper_patch
```
Comment (ethuleau) [inline lib/neutron_thirdparty/contrail]: You forget to set 'test_install_zookeeper_patch' function

File: contrail/zkServer.sh.patch [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
++++ /opt/stack/contrail/third_party/zookeeper-3.4.5/bin/zkServer.sh	2014-02-10 12:35:07.717084287 +0100
+@@ -97,6 +97,17 @@
+ 
+ _ZOO_DAEMON_OUT="$ZOO_LOG_DIR/zookeeper.out"
+ 
++JVM_OPTS=""
++if [ "$MAX_HEAP_SIZE"]; then
++    JVM_OPTS="$JVM_OPTS -Xms${MAX_HEAP_SIZE}"
++fi
++if [ "$MAX_HEAP_SIZE" ]; then
```
Comment (ethuleau) [inline contrail/zkServer.sh.patch]: Twice?

File: lib/neutron_thirdparty/contrail (hunk 1 above)
Comment (safchain) [inline lib/neutron_thirdparty/contrail]: Thx updated

**Comment by dsetia** [comment]
Cassandra change looks fine. It also gets rid of cassandra patch which is desirable to keep patches minimum. However, instead of introducing patch for zookeeper, could you look into using JVMFLAGS? This is already used in zkServer.sh. You could use something like this:

ZK_MAX_HEAP_SIZE=${ZK_MAX_HEAP_SIZE:-4G}
ZK_HEAP_NEWSIZE=${ZK_HEAP_NEWSIZE:-800M}
JVMFLAGS="-Xmx$ZK_MAX_HEAP_SIZE -Xmn$ZK_HEAP_NEWSIZE"

Then use JVMFLAGS while invoking zookeeper.

**Comment by safchain** [comment]
I'll keep the condition to set the zookeeper java env variables in order to keep the same behavior before this patch, I mean no default memory setting. I'll remove the cassandra patch and the zookeeper patch also.

**Comment by dsetia** [comment]
safchain, I would like to incorporate cassandra changes above. Can you generate a new pull request with cassandra fix alone for now? Also set the default values to 1G/200M to avoid hitting the memory condition?

**Comment by safchain** [comment]
Hi,

Ok I'll resend a new pull request soon

**Pull request closed without merging**